src/Services/Device.vala

```diff
@@ -231,29 +231,35 @@ public class Power.Services.Device : Object {
         }
         return (Power.Services.Device.Type) device.Type;
     }
-
     public string get_symbolic_icon_name_for_battery () {
         if (!is_a_battery) {
             return "preferences-system-power-symbolic";
         }
 
+        // Round to the nearest 5 percent and clamp to the range [20, 100]
+        int rounded_percentage = (int) (5 * Math.round (percentage / 5)).clamp (20, 100);
+
+        // Default icon name
         var icon_name = "battery";
 
+        // Set icon name based on percentage
         if (percentage > 10) {
-            // Round to the nearest 5 percent
-            // Clamp to 20. There is no 15. Make sure we don't have single px red line until < 10
-            var rounded_percentage = (int) (5 * Math.round (percentage / 5)).clamp (20, 100);
             icon_name += "-%i".printf (rounded_percentage);
         } else if (percentage > 0) {
             icon_name += "-10";
         } else {
             icon_name += "-0";
         }
 
+        // Add charging state if applicable
         if (is_charging) {
             icon_name += "-charging";
-        } else if (time_to_empty >= 0 && time_to_empty < 15 * 60) {
+        } else if (percentage == 0) {
+            // Only set battery-0 when the battery is empty (0%)
             icon_name = "battery-0";
+        } else if (time_to_empty >= 0 && time_to_empty < 15 * 60 && percentage > 0) {
+            // Prevent showing battery-0-symbolic if there's still battery left
+            icon_name = "battery-%i".printf (rounded_percentage);
```

Comment:
This will result in missing icon for those ranges below, I think it's 20

Reply:
What would be the case where it could fail? In a battery range between 0 and 20?

Reply:
Yeah IIRC the icons go from 0 to 20 and then every 5 or 10 after that. I can't remember and I'm AFK. But make sure to double check that :)

Reply:
I did the test with percentages less than 20% and they all work correctly.

![image](https://github.com/user-attachments/assets/5e363d1d-18de-4068-89e6-716e445ff435)